### ISSUE TITLE
Creating a cluster with templates that has invalid question format results in any other cluster creation to fail . 

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#21564]
Version: master-head (7/16/19)

Steps to reproduce the problem:
Create a template which results in template getting created with invalid question format like in https://github.com/rancher/rancher/issues/21563 and https://github.com/rancher/rancher/issues/21561.

After this failure to create a cluster , any cluster you attempt to create after running into this issue will fail to get created with the same error message.

Rancher server logs shows the following error repeatedly:

```
E0716 18:57:17.876537       6 reflector.go:134] github.com/rancher/norman/controller/generic_controller.go:175: Failed to list *v3.Cluster: json: cannot unmarshal string into Go struct field ClusterSpec.localClusterAuthEndpoint of type v3.LocalClusterAuthEndpoint
E0716 18:57:17.876731       6 reflector.go:134] github.com/rancher/norman/controller/generic_controller.go:175: Failed to list *v3.Cluster: json: cannot unmarshal string into Go struct field ClusterSpec.localClusterAuthEndpoint of type v3.LocalClusterAuthEndpoint

```

**Comment by alena1108** [comment]
@sangeethah validating the UI flow should be the first priority for this issue as UI would put a first line of defense on types validation. The API validation will be fixed too, as a second step. For now, if you run into this situation, removing the invalid cluster should correct things.

**Comment by prachidamle** [comment]
Yes removing the cluster that got created with incorrect field types, recovers the rancher server.
